### DISCUSSION TITLE
Installation with Homebrew

### DISCUSSION THREAD

**Discussion by merwok** [pism/pism#522] [General]
Hello!  tl;dr: mac instructions should mention homebrew commands

Long story: I am helping a scientist install PISM on a Mac laptop.  I run Debian on my machines and don’t use Macs, but I get a strong impression from other people (Python core developers, contributors and other Python developers) that Homebrew is the preferred developer tool to install system libraries.  So we took the instructions written for Macports and translated the package names to Homebrew, and could successfully install the PISM dependencies as well as Petsc (from homebrew too, not from source).
On one Mac laptop, running cmake (with CC options but no install prefix) then make worked straight up, but on another one the auto-detection didn’t happen and cmake errored out with a message about PETSC_LIB and PETSC_ARCH, which we didn’t have to set on the other machine.

It would be very appreciated if we could get help on how to let cmake find the Petsc installed by Homebrew.
And I think it would benefit even more people if the docs moved from Macports instructions to Homebrew 🙂

**Comment by ckhroulev** [comment]
Hi! Sorry about the late response -- somehow I missed your message until now. (Might have something to do with being on parental leave last summer...)

MacPorts-based instructions in the manual are from a few years back, when I used a Mac. I no longer use a Mac to work on PISM, so unfortunately I cannot easily update that section or test if it is up to date. I'll see what I can do, though.

**Comment by mankoff** [comment]
Note that for testing OS X stuff on Linux I have found this 1-button OS X in Virtualbox script to provide a working setup: https://github.com/myspaghetti/macos-virtualbox

**Reply by ckhroulev** [reply]
Thanks!

**Comment by merwok** [comment]


**Reply by mankoff** [reply]
It helps non-OS X developers test it, which helps OS X users.

**Comment by ckhroulev** [comment]
Setting up a macOS VM turned out to be too much of a pain, but I still made some progress.

Now we can install PISM 2.1 on macOS by running `brew tap pism/pism && brew install pism`. We test this setup using GitHub Actions and GitHub's macOS runners (currently: macOS 12 and macOS 13).

Run `brew install pism --HEAD` to install the current developmental version (`dev` branch).

To install tools and libraries required to build PISM, run `brew install --only-dependencies pism`. This should make it pretty easy to set up a development environment or prepare to build a different PISM version.

I will update [PISM's Homebrew formula](https://github.com/pism/homebrew-pism) and the manual for the 2.1.1 bug fix release.
